docs/design/README.md

```diff
@@ -0,0 +1,552 @@
+We call this *in-tree*, meaning that a variant can only be defined within Bottlerocket's git tree.
+Similarly, to add software to Bottlerocket, you need add it to the in-tree `packages` directory.
+
+A fundamental intention with variants is to allow for users to customize Bottlerocket for their unique use cases.
+To do so currently, however, requires forking and altering the Bottlerocket git repo.
+To stay up-to-date with core project would require frequent rebasing with conflicting changes in the core project.
+
+This document describes `twoliter`, a tool that will allow for the definition of a Bottlerocket variant externally from the main Bottlerocket project's git tree.
+Eventually `twoliter` will interface with another project underway to allow out-of-tree builds to define their own settings.
+Integration with the new settings system is out of scope for this document, and will be integrated with settings as both projects mature.
+
+### Glossary
+
+Here are a few of the terms we will use in this document:
+
+- **In-Tree**: Anything that exists in, or is defined by, the [main Bottlerocket git repository]
+- **Out-of-Tree Build (OOTB)**: User-customized builds of Bottlerocket defined by an external git repository owned by the user.
+- **Out-of-Tree Variants**: Synonymous with *Out of Tree Builds*.
+- **Builder**: A person who creates Bottlerocket variants using Twoliter.
+- **Kit**: A container image with RPM packages in it.
+
+[main Bottlerocket git repository]: https://github.com/bottlerocket-os/bottlerocket
+
+### Requirements
+
+The `twoliter` build tool should meet the following requirements.
+
+1. Binaries and metadata published for consumption by out-of-tree builds must be signed and automatically verified.
+2. Out-of-tree builds must support generating license attribution documents.
+3. All functionality necessary to build, test, and publish out-of-tree builds must be provided by a unified CLI.
+4. Maintainers must be able to define their own packages.
+5. Maintainers must be able to mix and match core project packages with their own.
+6. Maintainers must be able to replace a core project package with their own.
+7. Maintainers must be able to use core project packages without building them.
+8. Maintainers must be able to take and use packages from multiple sources.
+9. Maintainers must be able to fork and patch the core project in lieu of using the published binary artifacts.
+10. Maintainers must bring their own keys for all signing operations.
+11. Maintainers must be free to choose their own version numbering scheme.
+12. Maintainers must have a simple mechanism that can be used to update Bottlerocket dependencies to newer versions.
+13. Maintainers must be able to host their own copies of any required container images or source archives.
+14. Maintainers must be able to specify their own partition layouts (thank you @jfbette)
```

Comment:
It breaks up the flow a little to credit individuals in-line, but adding a "credits" section to the end for posterity seems fine.